BismarkUnchained/scenes/Player/Player.gd

```diff
@@ -77,10 +85,18 @@ func shoot():
 			get_node("/root/Main/UI/WeaponRecharge").value = 0
 	get_node("/root/Main/UI/WeaponRecharge").value = 100 * (get_node("/root/Main").time_elapsed - time_last_shot) / time_to_shoot
 
+func regen():
+	health += regen * (get_node("/root/Main").time_elapsed - last_regen)
+	if health > maxhealth:
+		health = maxhealth
+	last_regen = get_node("/root/Main").time_elapsed
```

Comment:
we should probably have a way to generalize this time delta (shifted time since last frame, not just wall time) - if you can think of something cleaner that'd be great, if not this is fine for now

Reply:
we can store in game time since last frame in main.gd probably